docs/API.md

```diff
@@ -0,0 +1,50 @@
+- **(HELP NEEDED)** `AsyncAPIDocument.isPublishingToChannels('<role>')` : boolean (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319))
+- `AsyncAPIDocument.messages()` : Message[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- `AsyncAPIDocument.schemas()` : Schema[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- `AsyncAPIDocument.servers()` : Server[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- `AsyncAPIDocument.server('<server name>')` : Server (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- `AsyncAPIDocument.json()` : string (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961), [800963792](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800963792))
+- `AsyncAPIDocument.securitySchemes()` : SecurityScheme[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+
+### Info
+- `Info.title()` : string (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319), [799598596](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799598596), [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407), [800963792](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800963792))
+- `Info.description()` : string (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319), [799598596](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799598596), [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407))
+- `Info.version()` : string (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319), [799598596](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799598596), [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407), [800963792](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800963792))
+
+### Server
+- `Server.name()` : string
+- `Server.protocol()` : string (outcome of comment(s): [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407))
+
+### SecurityScheme
+Empty for now.
+
+### Channel
+- `Channel.path()` : string
+- `Channel.messages()` : Message[]
+- **(HELP NEEDED)** `Channel.messagesPublishes('<role>')` : Message[]
+- **(HELP NEEDED)** `Channel.messagesSubscribes('<role>')` : Message[]
```

Comment:
What if we just avoid mentioning `publish` and `subscribe` here? Something like `messages() : Message[]`. To make it possible to filter by `publish` and `subscribe` operations, we can probably create another "entity" at the same level as Channel, Server, etc. but called Operation. That Operation object can have a `messages(): Message[]` too, where you only get the messages of this specific operation.

Reply:
Good point, we also discussed whether or not to add the `operation` as an entity. However, decided against it since we felt like it was more a property on the `channel` rather than an entity. We highlighted this in the description:
> So why are operation for example not an entity? Because we don't see the operation as needing an identity, it merely contains values we need associated with the Channel.

Reply:
But it's not. An operation is something tightly bound to the application. In the sentence:

```
My app publishes to channel X
```

I'd recognize the operation (verb) as a trait of the application (subject) and the channel is a mere addition. I believe all the weird things happening here and below are because of the lack of an Operation entity.

Reply:
I see the operation as the real intent of the user (Publish? Subscribe? Whatever?) linked optionally to some message(s).

Your suggestion is to create a new `Operation` entity. However, an Operation won't ever exist without, at least, a channel and a message (please correct me if I'm wrong).
Why don't we, instead, give channels and messages to the users, and the ability to filter those by operation (as a concept rather than an entity?).

Reply:
Yea, think you might be right here.

cc. @smoya

Reply:
> our suggestion is to create a new Operation entity. However, an Operation won't ever exist without, at least, a channel and a message (please correct me if I'm wrong).

An operation would not exist without an Application, a Channel, and a Message. That's the problem. It's linked to all the other entities (except Server, at least not directly). And to make it even more complex, an "Application" here can mean different things: the app described in the AsyncAPI file or the client application. It's a directed graph. That's why you need the application part, to understand the direction of the graph.

```
Application --> publish --> Message --> Channel
```

But who's the Application here? If it's the "provider", then the graph above talks about behavior. If it's the "client", the graph above talks about capability (e.g., a client **can** publish).

We can solve this problem by creating more "verbs" to make it clearer:

```
Application --> performs/allows --> Operation{type, subject, message, channel}
```

Where `type` can be `publish`, `subscribe`, or maybe even `request`. And `subject` can be `provider`/`application` or `client`. I don't foresee more "subjects" appearing but I do foresee more "types" appearing over time, like `query` and `mutation`.

> Why don't we, instead, give channels and messages to the users, and the ability to filter those by operation (as a concept rather than an entity?)

Who are "the users" here? What's going to be the point of view of the operation? From the client or the provider?

Reply:
> But who's the Application here? 

That question is repeated several times during our research (but also formulated by other community members within issues) made us introduce the concept of the `role`. We talk about the role here.

> We can solve this problem by creating more "verbs" to make it clearer:

Then you suggest to rather convert Operation as a first-class Entity behaving like the unit of work an application or a  client can perform. 
As a note, you are talking about possible future changes within the spec, which I see are not that far from the first change I emulated in https://github.com/asyncapi/shape-up-process/issues/93#issuecomment-803973100

In conclusion, this will mean our spec would look more Operation-oriented, which in fact it aligns with the average use-case (to describe async operations?).

I have few technical questions regarding the Parser API:

- Would you create a method per operation verb and role (or whatever we wanna call it)? You suggested `AsyncAPIDocument.clientPublishableChannels()` and `AsyncAPIDocument.publishableChannels()`. Would that mean in the future we would need to introduce methods such as `AsyncAPIDocument.clientRequestableChannels()` ?
- Would you introduce similar intents but for Messages and Operations? For example:
  - `AsyncAPIDocument.publishableMessages()`
  -  `AsyncAPIDocument.publishableOperations()`
  -  `AsyncAPIDocument.clientSubscribableOperations()`
  -  `AsyncAPIDocument.subscribableOperations()`
  -  `AsyncAPIDocument.requestableOperations()` (future)
- Similar to the above, but for intents on messages, channels and operations in an intent-fashion shape? For example:
    - `message.clientPublishableChannels()`
    - `channel.PublishableMessages()`
    - `operation.messages()`

Reply:
Yes, it will —most probably— become more operation-oriented.

And regarding your questions, my answers are:

- Yes
- Yes
- Don't have an opinion :)

Reply:
@fmvilas Changes applied here https://github.com/asyncapi/parser-js/pull/263/commits/6da212318e582f0927685f25936d882ff6c529a5

Reply:
Only the API changed not the description*